lib/streamlit/elements/lib/built_in_chart_utils.py

```diff
@@ -0,0 +1,837 @@
+    )
+
+
+def _parse_generic_column(
+    df: pd.DataFrame, column_or_value: Any
+) -> tuple[str | None, Any]:
+    if isinstance(column_or_value, str) and column_or_value in df.columns:
+        column_name = column_or_value
+        value = None
+    else:
+        column_name = None
+        value = column_or_value
+
+    return column_name, value
+
+
+def _parse_x_column(df: pd.DataFrame, x_from_user: str | None) -> str | None:
+    if x_from_user is None:
+        return None
+
+    elif isinstance(x_from_user, str):
+        if x_from_user not in df.columns:
+            raise StreamlitColumnNotFoundError(df, x_from_user)
+
+        return x_from_user
+
+    else:
+        raise StreamlitAPIException(
+            "x parameter should be a column name (str) or None to use the "
+            f" dataframe's index. Value given: {x_from_user} "
+            f"(type {type(x_from_user)})"
+        )
+
+
+def _parse_y_columns(
+    df: pd.DataFrame,
+    y_from_user: str | Sequence[str] | None,
+    x_column: str | None,
+) -> list[str]:
+    y_column_list: list[str] = []
+
+    if y_from_user is None:
+        y_column_list = list(df.columns)
+
+    elif isinstance(y_from_user, str):
+        y_column_list = [y_from_user]
+
+    elif type_util.is_sequence(y_from_user):
+        y_column_list = list(str(col) for col in y_from_user)
+
+    else:
+        raise StreamlitAPIException(
+            "y parameter should be a column name (str) or list thereof. "
+            f"Value given: {y_from_user} (type {type(y_from_user)})"
+        )
+
+    for col in y_column_list:
+        if col not in df.columns:
+            raise StreamlitColumnNotFoundError(df, col)
+
+    # y_column_list should only include x_column when user explicitly asked for it.
+    if x_column in y_column_list and (not y_from_user or x_column not in y_from_user):
+        y_column_list.remove(x_column)
+
+    return y_column_list
+
+
+def _get_opacity_encoding(
+    chart_type: ChartType, color_column: str | None
+) -> alt.OpacityValue | None:
+    import altair as alt
+
+    if color_column and chart_type == ChartType.AREA:
+        return alt.OpacityValue(0.7)
+
+    return None
+
+
+def _get_axis_config(df: pd.DataFrame, column_name: str | None, grid: bool) -> alt.Axis:
+    import altair as alt
+    from pandas.api.types import is_integer_dtype
+
+    if column_name is not None and is_integer_dtype(df[column_name]):
+        # Use a max tick size of 1 for integer columns (prevents zoom into float numbers)
+        # and deactivate grid lines for x-axis
+        return alt.Axis(tickMinStep=1, grid=grid)
+
+    return alt.Axis(grid=grid)
+
+
+def _maybe_melt(
+    df: pd.DataFrame,
+    x_column: str | None,
+    y_column_list: list[str],
+    color_column: str | None,
+    size_column: str | None,
+) -> tuple[pd.DataFrame, str | None, str | None]:
+    """If multiple columns are set for y, melt the dataframe into long format."""
```

Comment:
nit: could this comment be extended with an example? I am not sure what it means to melt a dataframe into long format

Reply:
Added more info to the comment